lpc55_sign/src/bin/lpc55_sign.rs

```diff
@@ -137,15 +157,293 @@ fn main() -> Result<()> {
         }
-        ImageType::EccImage {
+        Command::EccImage {
             src_bin,
             priv_key,
             dest_bin,
         } => {
             sign_ecc::ecc_sign_image(&src_bin, &priv_key, &dest_bin)?;
             println!("Done! ECC image written to {:?}", &dest_bin);
         }
+        Command::VerifySignedImage {
+            src_cmpa,
+            src_cfpa,
+            src_img,
+            verbose,
+        } => {
+            println!("=== CMPA ====");
+            let cmpa = {
+                let mut cmpa_bytes = [0u8; 512];
+                let mut cmpa_file = std::fs::File::open(src_cmpa)?;
+                cmpa_file.read_exact(&mut cmpa_bytes)?;
+                CMPAPage::from_bytes(&cmpa_bytes)?
+            };
+
+            let boot_cfg = cmpa.get_boot_cfg()?;
+            let secure_boot_cfg = cmpa.get_secure_boot_cfg()?;
+            let cc_socu_pin = cmpa.get_cc_socu_pin()?;
+            let cc_socu_dflt = cmpa.get_cc_socu_dflt()?;
+
+            if verbose {
+                println!("{:#?}", boot_cfg);
+                println!("{:#?}", secure_boot_cfg);
+                println!("{:#?}", cc_socu_pin);
+                println!("{:#?}", cc_socu_dflt);
+                println!("ROTKH: {:}", hex::encode(cmpa.rotkh));
+            }
+            println!("No CMPA verification implemented");
+
+            println!("=== CFPA ====");
+            let cfpa = {
+                let mut cfpa_bytes = [0u8; 512];
+                let mut cfpa_file = std::fs::File::open(src_cfpa)?;
+                cfpa_file.read_exact(&mut cfpa_bytes)?;
+                CFPAPage::from_bytes(&cfpa_bytes)?
+            };
+
+            let rkth_revoke = cfpa.get_rkth_revoke()?;
+
+            if verbose {
+                println!("Version: {:x}", cfpa.version);
+                println!("Secure FW Version: {:x}", cfpa.secure_firmware_version);
+                println!("Non-secure FW Version: {:x}", cfpa.ns_fw_version);
+                println!("Image key revoke: {:x}", cfpa.image_key_revoke);
+                println!("{:#?}", rkth_revoke);
+            }
+            println!("No CFPA verification implemented");
+
+            println!("=== Image ====");
+            let image = std::fs::read(src_img)?;
+            let image_len = u32::from_le_bytes(image[0x20..0x24].try_into().unwrap());
+            let image_type = BootField::unpack(image[0x24..0x28].try_into().unwrap())?;
+
+            let load_addr = u32::from_le_bytes(image[0x34..0x38].try_into().unwrap());
+            let is_plain = image_type.img_type == EnumCatchAll::Enum(BootImageType::PlainImage);
+
+            if verbose {
+                println!("image length: {image_len:#x} ({image_len})");
+                println!("image type: {image_type:#?}");
+                println!("load address: {load_addr:#x}");
+            }
+            if is_plain && load_addr != 0 {
+                println!("⚠️  Load address is non-0 in a non-plain image");
```

Comment:
I like these emoji; however I think they are unlikely to work properly on xterm (which I'm still using for some reason) and consoles of things outside of X (which iirc the manufacturing stations are). Maybe test how it degrades.